tests/Microsoft.Identity.Test.Unit/RequestsTests/LongRunningOnBehalfOfTests.cs

```diff
@@ -928,6 +928,48 @@ await app.InitiateLongRunningProcessInWebApi(TestConstants.s_scope, TestConstant
             }
         }
 
+        [TestMethod]
+        public async Task LongRunningObo_WithClaims_TestAsync()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                httpManager.AddInstanceDiscoveryMockHandler();
+
+                // First mock handler for InitiateLongRunningProcessInWebApi
+                AddMockHandlerAadSuccess(httpManager);
+
+                var cca = BuildCCA(httpManager);
+
+                // We'll track the OBO cache key
+                string oboCacheKey = null;
+
+                // STEP 1: Initiate LRO OBO with claims
+                var initResult = await cca
+                    .InitiateLongRunningProcessInWebApi(TestConstants.s_scope, TestConstants.DefaultAccessToken, ref oboCacheKey)
+                    .WithClaims(TestConstants.Claims)
+                    .ExecuteAsync()
+                    .ConfigureAwait(false);
+
+                // Because we used claims, MSAL sets CacheRefreshReason.WithClaims, skipping the cached token
+                Assert.AreEqual(TokenSource.IdentityProvider, initResult.AuthenticationResultMetadata.TokenSource);
+                Assert.AreEqual(CacheRefreshReason.NotApplicable, initResult.AuthenticationResultMetadata.CacheRefreshReason);
```

Comment:
the comment above says MSAL sets the reason to `CacheRefreshReason.WithClaims`, however, this is checking for CacheRefreshReason.NotApplicable?

Reply:
for the obo flow yes, the first call is not applicable.